libpod/container_log_linux.go

```diff
@@ -80,39 +80,25 @@ func (c *Container) readFromJournal(ctx context.Context, options *logs.LogOption
 		return fmt.Errorf("adding filter to journald logger: %v: %w", match, err)
 	}
 
-	if err := journal.SeekHead(); err != nil {
-		return err
-	}
-	// API requires Next() immediately after SeekHead().
-	if _, err := journal.Next(); err != nil {
-		return fmt.Errorf("next journal: %w", err)
-	}
-
-	// API requires a next|prev before getting a cursor.
-	if _, err := journal.Previous(); err != nil {
-		return fmt.Errorf("previous journal: %w", err)
-	}
-
-	// Note that the initial cursor may not yet be ready, so we'll do an
-	// exponential backoff.
-	var cursor string
-	var cursorError error
-	var containerCouldBeLogging bool
-	for i := 1; i <= 3; i++ {
-		cursor, cursorError = journal.GetCursor()
-		hundreds := 1
-		for j := 1; j < i; j++ {
-			hundreds *= 2
+	if options.Since.IsZero() {
+		if err := journal.SeekHead(); err != nil {
+			return err
 		}
-		if cursorError != nil {
-			time.Sleep(time.Duration(hundreds*100) * time.Millisecond)
-			continue
+	} else {
+		// seek based on time which helps to reduce unnecessary event reads
+		if err := journal.SeekRealtimeUsec(uint64(options.Since.UnixMicro())); err != nil {
+			return err
 		}
-		break
 	}
-	if cursorError != nil {
-		return fmt.Errorf("initial journal cursor: %w", cursorError)
+
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	if err := c.syncContainer(); err != nil {
+		return err
 	}
+	// the initial could be logging state must be correct, we cannot rely on the start event being still in the journal
```

Comment:
I don't understand this sentence, but will not block over it. I will trust that more knowledgeable people will either understand it, or block the PR requesting clarification.

Reply:
The "not being in the journal" part confused me as well but the test gave clarity.  It's probably worth mentioning that `--since` may be set after the start event, so we need to take the initial container state into account before reading the journal.

Reply:
--since just triggers it more easily but it is actually a general problem, if your container runs long enough the journal will be rotated at some point and you can loose the container start event.

Reply:
I made it more clear when/how it can happen.